server/src/main/java/org/opensearch/index/compositeindex/CompositeIndexSettings.java

```diff
@@ -37,12 +39,22 @@ public class CompositeIndexSettings {
         Setting.Property.Dynamic
     );
 
+    /**
+     * This sets the max flush threshold size for composite index
+     */
+    public static final Setting<ByteSizeValue> COMPOSITE_INDEX_MAX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING = Setting.byteSizeSetting(
+        "indices.composite_index.translog.max_flush_threshold_size",
+        new ByteSizeValue(512, ByteSizeUnit.MB),
+        new ByteSizeValue(128, ByteSizeUnit.MB),
+        new ByteSizeValue(Long.MAX_VALUE, ByteSizeUnit.BYTES),
+        Setting.Property.NodeScope
```

Comment:
I think we may have a way to make it dynamic by using nodeSettings in IndexSettings object to be fetched in `setTranslogFlushThresholdSize` method for validation. Can we try that out?

Reply:
This seems not possible, nodeSettings seems to be different from cluster settings. Some of them getting initialized from yml, some of them in node.java.